app/assets/javascripts/features/machines.js

```diff
@@ -0,0 +1,49 @@
+const buttons = [...document.querySelectorAll('button')]
+const solutions = [...document.querySelectorAll('.problem__solutions')]
+const headers = [...document.querySelectorAll('.device__problems header')]
```

Comment:
So first time working with a Nodelist, but without the wrapped array, it renders as a nodelist. And `.map()` doesn't work with nodelists but only arrays...